relayer-cli/src/commands/keys/add.rs

```diff
@@ -47,17 +49,18 @@ impl Runnable for KeysAddCmd {
 
         let opts = match self.validate_options(&config) {
             Err(err) => {
-                status_err!("invalid options: {}", err);
-                return;
+                return Output::with_error().with_result(json!(err)).exit();
             }
             Ok(result) => result,
         };
 
         let res: Result<String, Error> = add_key(opts).map_err(|e| Kind::Keys.context(e).into());
 
         match res {
-            Ok(r) => status_info!("key add result: ", "{:?}", r),
-            Err(e) => status_info!("key add failed: ", "{}", e),
+            Ok(r) => Output::with_success().with_result(json!(r)).exit(),
+            Err(e) => Output::with_error()
+                .with_result(json!(format!("{}", e)))
+                .exit(),
```

Comment:
I wonder if we could introduce a couple helpers to streamline such code a bit, eg.

```suggestion
            Ok(r) => Output::success(json!(r)).exit(),
            Err(e) => Output::error(json!(format!("{}", e))).exit(),
```

Reply:
Additionally, what do you think about changing the type of `success`/`error`/`with_result` to

```rust
pub fn with_result(mut self, res: impl Serialize) -> Self {
        self.result.push(serde_json::to_value(res).unwrap()); // same unwrap as in `json!` macro
        self
    }
```

which would result into

```suggestion
            Ok(r) => Output::success(r).exit(),
            Err(e) => Output::error(format!("{}", e)).exit(),
```

Having to call `unwrap` on the result of `serde_json::to_value` is unfortunate, but that's actually what [the `json!` macro does under the hood](https://docs.rs/serde_json/1.0.61/src/serde_json/macros.rs.html#278) so perhaps it's okay?

Also note that this would still allow passing a custom JSON object with the `json!` macro:

```rust
Output::success(json!({ "hello": "world" })).exit()
```